### PR TITLE
Optim

### PR DESCRIPTION
**Pull request by eolivi**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -76,7 +76,7 @@ On Mac OS X, you'll need `CMake <http://www.cmake.org>`_ (install it with `Homeb
 
     *optionally*::
 
-    $ brew install hdf5 && brew install libmatio --with-hdf5
+    $ brew install hdf5 && brew install libmatio
```
Comment (agramfort) [inline README.rst]: please revert. it's in my other PR :)

Comment (eolivi) [inline README.rst]: yes I've seen.
I'll wait for other comments ?

**Comment by agramfort** [comment]
nope that's it for me

**Comment by agramfort** [comment]
appveyor is not happy

**Comment by eolivi** [comment]
Making OperatorD Optimzed and parallel is a big optimization as it reduces the time for HM-Head3 from 45s to 18s

File: build_tools/travis/before_install.sh [hunk 2]
```diff
@@ -28,6 +28,10 @@ if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
         brew install vtk
     fi
 
+    if [[ "$USE_OMP" == 1 ]]; then
+        brew install llvm
+    fi
+
```
Comment (eolivi) [inline build_tools/travis/before_install.sh]: @agramfort  please is that the correct formula we need to build with openMP ? or clang support it by default ?

Comment (agramfort) [inline build_tools/travis/before_install.sh]: I have no idea.

Before clang we had gcc 4.2 on mac and I did the previous release using the intel compiler anyway...

**Comment by agramfort** [comment]
so you removed the non-optimized D operator?

File: OpenMEEG/libs/OpenMEEG/include/operators.h [hunk 3]
```diff
@@ -394,7 +315,13 @@ namespace OpenMEEG {
         //
 
         unsigned i = 0; // for the PROGRESSBAR
-        for (Mesh::const_iterator tit1=m1.begin();tit1!=m1.end();++tit1) {
+        #pragma omp parallel for
+        #ifndef OPENMP_3_0
+        for (int i1=0; i1 < m1.size(); ++i1) {
+            const Mesh::const_iterator tit1 = m1.begin()+i1;
```
Comment (agramfort) [inline OpenMEEG/libs/OpenMEEG/include/operators.h]: is this the improvement?

Comment (eolivi) [inline OpenMEEG/libs/OpenMEEG/include/operators.h]: yes !

Comment (eolivi) [inline OpenMEEG/libs/OpenMEEG/include/operators.h]: yes

File: build_tools/travis/before_install.sh [hunk 4]
```diff
@@ -28,6 +28,10 @@ if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
         brew install vtk
     fi
 
+    if [[ "$USE_OMP" == 1 ]]; then
+        brew install llvm
```
Comment (agramfort) [inline build_tools/travis/before_install.sh]: ok I tested. It does the job but you need to change the compiler that cmake is using.

for me I need to set:

CPP = /usr/local/opt/llvm/bin/clang
CPPFLAGS = -I/usr/local/opt/llvm/include -fopenmp
LDFLAGS = -L/usr/local/opt/llvm/lib

this being said we should not ship binaries with clang-omp / llvm on mac as it will break on most systems that don't brew setup properly.

We should ship non-openmp binaries on mac and update the brew formula to add an openmp option.

So to sum up:
- setup/update only one travis build so it uses openmp using brew setting up properly all env vars.
- upload to gforge non-openmp binaries for our users

Comment (eolivi) [inline build_tools/travis/before_install.sh]: >CMake Error at /usr/local/Cellar/cmake/3.8.2/share/cmake/Modules/CMakeTestCCompiler.cmake:51 (message):
  The C compiler
  "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"
  is not able to compile a simple test program.
  It fails with the following output:
   Change Dir: /Users/travis/build/openmeeg/openmeeg/build/OpenMEEG/build/CMakeFiles/CMakeTmp
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
  -fopenmp=libomp -w -o CMakeFiles/cmTC_f4998.dir/testCCompiler.c.o -c
  /Users/travis/build/openmeeg/openmeeg/build/OpenMEEG/build/CMakeFiles/CMakeTmp/testCCompiler.c
  clang: error: unsupported argument 'libomp' to option 'fopenmp='

for now I have that, while I've set:
        export CXX="/usr/local/opt/llvm/bin/clang"
        export CPPFLAGS="$CPPFLAGS -I/usr/local/opt/llvm/include -fopenmp"
        export LDFLAGS="$LDFLAGS -L/usr/local/opt/llvm/lib"

I'm trying on our ci mac machine... but it takes ages to brew llvm

**Comment by agramfort** [comment]
travis passes with this?

File: build_tools/travis/before_install.sh (hunk 4 above)
Comment (agramfort) [inline build_tools/travis/before_install.sh]: you're still using xcode clang and not the one from brew. See

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang

in travis log.

you need to set flags for C and not only CPP files.

**Comment by eolivi** [comment]
arf... cannot try this on my Mac from ci, since I cannot install llvm (at least with brew)...
It says that MacOsX 10.9 is too old for this...

**Comment by agramfort** [comment]
ok let try

**Comment by agramfort** [comment]
I tried adding

        export CC="/usr/local/opt/llvm/bin/clang"
        export CXX="/usr/local/opt/llvm/bin/clang"
        export CPPFLAGS="$CPPFLAGS -I/usr/local/opt/llvm/include -fopenmp"
        export CFLAGS="$CFLAGS -I/usr/local/opt/llvm/include -fopenmp"
        export LDFLAGS="$LDFLAGS -L/usr/local/opt/llvm/lib"

but I get a link error. Can you try with travis?

**Comment by agramfort** [comment]
travis is not happy at all here.

this PR is the top priority.

I would leave out OMP for mac to converge quickly.

I have still hope to have a prelease before OHBM

**Comment by eolivi** [comment]
I think it is okay now. with OpenMP.
The error was
> export CXX="/usr/local/opt/llvm/bin/clang"

which should be
> export CXX="/usr/local/opt/llvm/bin/clang++"

**Comment by eolivi** [comment]
good to review/merge . it works with OpenMP
previously:
HM-HeadNNc2 ...................................   Passed    8.25 sec
whereas now:
HM-HeadNNc2 ...................................   Passed    3.86 sec

**Comment by agramfort** [comment]
great !

thanks @eolivi 

do we have binaries to test on windows and linux now?
